File: apps/android/app/src/main/java/com/rakizz/student/domain/model/Quiz.kt
package com.rakizz.student.domain.model

data class QuizQuestion(
    val id: String,
    val questionText: String,
    val options: List<String>
)

data class QuizReviewItem(
    val id: String,
    val questionText: String,
    val options: List<String>,
    val correctAnswer: String,
    val explanation: String,
    val sourceChunkSnippet: String?
)

data class Quiz(
    val id: String,
    val title: String,
    val description: String,
    val score: Int?,
    val totalQuestions: Int,
    val materialId: String = "",
    val questions: List<QuizQuestion> = emptyList(),
    val passed: Boolean? = null,
    val reviewData: List<QuizReviewItem> = emptyList()
)

File: packages/contracts/shared_python/schemas/quiz.py
from pydantic import BaseModel
from typing import List, Dict, Optional
from uuid import UUID
from enum import Enum


class DifficultyLevel(str, Enum):
    EASY = "EASY"
    MEDIUM = "MEDIUM"
    HARD = "HARD"

    # this is what the app will use now
    # student will not choose the level
    MIXED = "MIXED"


class QuizGenerateRequest(BaseModel):
    material_id: UUID
    difficulty: DifficultyLevel


class QuizQuestionPublic(BaseModel):
    id: UUID
    question_text: str
    options: List[str]

    class Config:
        from_attributes = True


class QuizQuestionReview(BaseModel):
    id: UUID
    question_text: str
    options: List[str]
    correct_answer: str
    explanation: str
    source_chunk_snippet: Optional[str] = None

    class Config:
        from_attributes = True


class QuizSetPublicResponse(BaseModel):
    id: UUID
    material_id: UUID
    questions: List[QuizQuestionPublic]

    class Config:
        from_attributes = True


class QuizAttemptCreate(BaseModel):
    answers: Dict[UUID, str]


class QuizAttemptResultResponse(BaseModel):
    id: UUID
    quiz_set_id: UUID
    student_id: UUID
    score: float
    passed: bool
    review_data: List[QuizQuestionReview]

    class Config:
        from_attributes = True